docs/GUIDE.md

```diff
@@ -0,0 +1,142 @@
+      "org.example.MySerializable" = circe-json
+    }
+  }
+}
+```
+where `org.example.ExampleSerializer` is the FQCN of your Serializer which extends CirceAkkaSerializer and `org.example.MySerializable` is the FQCN of your top-level serializable type (trait / abstract class) that is extended by Messages / Events / States which `org.example.ExampleSerializer` should serialize.
+
+Also, there are few optional configurations that you can set if you want to change the default behavior of Circe Akka Serializer:
+- `org.virtuslab.ash.circe.verbose-debug-logging` = `on` / `off`
+
+`on` enables debug logs for each serialization / deserialization -> prints info about what has been serialized/deserialized and how much time it took. Default is `off`.<br><br>
+
+- `org.virtuslab.ash.circe.enable-missing-codecs-check` = `true` / `false`
+
+`true` enables additional check for possible missing codec registrations (more info about this problem in [README](../README.md)). This is in general checked by the [Codec Registration Checker Compiler Plugin](#codec-registration-checker-compiler-plugin-guide) during compilation, so the default is `false`. However, if you use only Circe Akka Serializer without compiler plugins - this check should be enabled. Be aware that enabling this check might have a slight negative impact on Serializer's performance.<br><br>
+
+- `org.virtuslab.ash.circe.compression.algorithm` = `gzip` / `off`
+
+`gzip` enables compression of payloads before serialization using the [Gzip](https://www.gnu.org/software/gzip) algorithm. Default is `off` (compression disabled). Choosing `gzip` altogether with setting proper `compress-larger-than` value (explained below) enables compression of payloads.<br><br>
+
+- `org.virtuslab.ash.circe.compression.compress-larger-than` = `64 KiB` _(example value)_
+
+If `org.virtuslab.ash.circe.compression.algorithm` = `gzip` and `compress-larger-than` **value** is larger than `0 KiB` &mdash; each payload greater than the **value** will be compressed before serialization. Default **value** is `32 KiB`. If `org.virtuslab.ash.circe.compression.algorithm` = `off` &mdash; payloads will not be compressed, regardless of the chosen `compress-larger-than` value.<br><br>
+**Note** &mdash; Circe Akka Serializer will perform **deserialization** properly for both compressed and not compressed payloads regardless of both `org.virtuslab.ash.circe.compression` configurations (Serializer will recognize whether the payload has been compressed programmatically). These settings matter only for choosing type of serialization.
+
+### Annotations Guide
+ASH compiler plugins are based on usages of two annotations: [@SerializabilityTrait](#SerializabilityTrait) and [@Serializer](#Serializer). Thus, before enabling compiler plugins, make sure that you are using these two annotations properly in the project / module where plugins will do their work. In order to have access to these annotations, add this part of ASH to library dependencies:
```

Comment:
As below... _before enabling compiler plugins_... but users will rarely need to enable the compiler plugins explicitly, right? they'll instead just rely on ASH sbt plugin to apply these plugins

Reply:
yeah, this part will be fixed in upcoming commit